hydromt/gis_utils.py

```diff
@@ -311,17 +311,21 @@ def meridian_offset(ds, bbox=None):
     if (
         ds.raster.crs is None
         or ds.raster.crs.is_projected
-        or not np.isclose(e - w, 360)
+        or not np.isclose(e - w, 360)  # grid should span 360 degrees!
     ):
         raise ValueError("The method is only applicable to global geographic CRS")
     x_name = ds.raster.x_dim
     lons = np.copy(ds[x_name].values)
-    if bbox is not None and bbox[0] < w and bbox[0] < -180:  # 180W - 180E > 360W - 0W
-        lons = np.where(lons > 0, lons - 360, lons)
-    elif bbox is not None and bbox[2] > e and bbox[2] > 180:  # 180W - 180E > 0E-360E
-        lons = np.where(lons < 0, lons + 360, lons)
-    elif np.isclose(e, 360):  # 0W - 360E > 180W - 180E (allow for rounding errors)
-        lons = np.where(lons > 180, lons - 360, lons)
+    if bbox is not None:  # bbox west and east
+        bbox_w, bbox_e = bbox[0], bbox[2]
+    else:  # global west and east in case of no bbox
+        bbox_w, bbox_e = -180, 180
+    if bbox_w < w:  # shift lons east of x0 by 360 degrees west
+        x0 = 180 if bbox_w >= -180 else 0
+        lons = np.where(lons > max(bbox_e, x0), lons - 360, lons)
+    elif bbox_e > e:  # shift lons west of x0 by 360 degrees east
+        x0 = -180 if bbox_e <= 180 else 0
+        lons = np.where(lons < min(bbox_w, x0), lons + 360, lons)
```

Comment:
I see the logic of this and its sound. I was wondering one thing though, which may be completely unrealistic, but what if the data is [-90, -85, 270, 85] (I know its a bit ridiculous) and one supplies a bbox from [-2, -5, 2, 5]. The data will not be shifted as the bbox falls completely within the data, but the bbox is most likely normal while the data is off. Just wondering.

Reply:
> but the bbox is most likely normal while the data is off. Just wondering.

What do you mean by this? We assume the coordinates are correct and we apply the shift to be able to clip the correct portion of the data. So the behaviour is correct here, right?

Reply:
Well, I said the logic is sound. 
I was until now more under the impression that the method was meant to make sure the global data is not situated in a weird way. That's why I mentioned that hypothetical because for that case it could present unexpected results. 
Was wondering what your opinion is on that.

Reply:
But maybe let's have a talk about this later.

Reply:
I've updated the method docstring to clarify what it is suppose to do.